### PR TITLE
Gateway hosts may contain wildcards

### PR DESCRIPTION
**Pull request by rosenhouse**
Hacking away at #3410

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v1/gateway.go [hunk 1]
```diff
@@ -166,47 +180,56 @@ func buildGatewayVirtualHosts(configStore model.IstioConfigStore, node model.Pro
...
+		if stringSliceContains(matchCriteria, hosts) {
+			return []string{matchCriteria}
+		}
+	}
+
+	matchingHosts := []string{}
+	for _, host := range hosts {
+		if strings.HasSuffix(host, matchCriteria[1:]) {
+			matchingHosts = append(matchingHosts, host)
+		}
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v1/gateway.go]: Are we sticking to prefix based matches only? Or should we allow regex matches? Or wildcards in any position like foo.*.com ?
I am okay with the prefix based match as a first cut, but wondering if such use cases exist in cf world.

Comment (rosenhouse) [reply pilot/pkg/proxy/envoy/v1/gateway.go]: No use cases that I know of....

**Pull request merged**